convert/convert/converters/labelstudio/badgerdoc_downloader.py

```diff
@@ -0,0 +1,132 @@
+import tempfile
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+from botocore.client import BaseClient
+from botocore.exceptions import ClientError
+from fastapi import HTTPException
+
+from convert.converters.base_format.models import annotation_practic
+from convert.converters.base_format.models.annotation import (
+    BadgerdocAnnotation,
+)
+from convert.converters.base_format.models.manifest import Manifest
+from convert.converters.base_format.models.tokens import Page
+from convert.converters.labelstudio.annotation_converter_practic import (
+    AnnotationConverterToTheory,
+)
+from convert.logger import get_logger
+from convert.models.common import S3Path
+
+LOGGER = get_logger(__file__)
+LOGGER.setLevel("DEBUG")
+
+
+class BadgerdocDownloader:
+    def __init__(
+        self,
+        s3_client: BaseClient,
+        s3_input_tokens: S3Path,
+        s3_input_manifest: S3Path,
+    ):
+        self.s3_input_tokens = s3_input_tokens
+        self.s3_input_manifest = s3_input_manifest
+        self.s3_client = s3_client
+
+    def download(
```

Comment:
It's not only downloading relevant files, it also parses it.
I would call it "download_and_parse" or any other descriptive name